frontend/src/views/CellCards/components/CellCardSearchBar/index.tsx

```diff
@@ -0,0 +1,136 @@
+import { useState } from "react";
+import { TextField } from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import InputAdornment from "@mui/material/InputAdornment";
+import { SectionItem, SectionTitle, StyledAutocomplete } from "./style";
+import { ROUTES } from "src/common/constants/routes";
+import { useCellTypes } from "src/common/queries/cellCards";
+import { useRouter } from "next/router";
+
+interface CellType {
+  id: string;
+  label: string;
+}
+
+export default function CellCardSearchBar(): JSX.Element {
+  const router = useRouter();
+  const { data: cellTypes } = useCellTypes();
+
+  const [open, setOpen] = useState(false);
+  const [inputValue, setValue] = useState("");
+
+  // Used for keyboard navigation for cell type search
+  const [highlightedCellType, setHighlightedCellType] =
+    useState<CellType | null>(null);
+
+  const handleFocus = () => {
+    setOpen(true);
+  };
+
+  const handleBlur = () => {
+    setOpen(false);
+  };
+
+  function changeCellType(cellTypeId: string) {
+    if (cellTypeId) {
+      router.push(`${ROUTES.CELL_CARDS}/${cellTypeId.replace(":", "_")}`);
+      document.getElementById("cell-cards-search-bar")?.blur();
+      setOpen(false);
+    }
+  }
+
+  return (
+    <div data-testid="cell-card-search-bar">
+      <StyledAutocomplete
+        open={open}
+        value={inputValue}
+        onChange={() => {
+          // Clears the input after selection
+          setValue("");
+        }}
+        onKeyDown={(event) => {
+          if (highlightedCellType && event.key === "Enter") {
+            changeCellType(highlightedCellType.id);
+          }
+        }}
+        onHighlightChange={(_, value) => {
+          const cellType = value as CellType;
+          setHighlightedCellType(cellType);
+        }}
+        disablePortal
+        id="cell-cards-search-bar"
+        options={cellTypes ?? []}
+        groupBy={(option) => {
```

Comment:
@ashin-czi Since you last looked at this, I added `groupBy` and `renderGroup` properties to add the "CELL TYPES" header (this is for when we will be able to search by both tissues and cell types).

<img width="475" alt="image" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/16548075/092fb8fa-b086-4162-a5ca-7154fce4e452">